spyder/plugins/ipythonconsole.py

```diff
@@ -927,6 +944,7 @@ def create_new_client(self, give_focus=True, path=''):
                                      "<tt>conda install ipykernel</tt>"))
                 return
 
+        client.update_cwd.connect(self.set_working_directory)

```

Comment:
Please move this line to `register_client` and change it like this

    shellwidget.sig_change_cwd.connect(self.set_working_directory)